test/README.md

```diff
@@ -0,0 +1,49 @@
+| `setupTimeoutMinutes` || The maximum amount of time, in minutes, test setup should take. This includes setup for all tests, for the tests in a fixture, or for a single test. If this time is exceeded, the associated test(s) will fail with a timeout error. If not given, the default value is `5`. |
+| `teardownTimeoutMinutes` || The maximum amount of time, in minutes, test teardown should take. This includes teardown for all tests, for the tests in a fixture, or for a single test. If this time is exceeded, the associated test(s) will fail with a timeout error. If not given, the default value is `2`. |
+| `tempSensorImage` || Docker image to pull/use for the 'TempSensor' module (see the test 'TempSensor'). |
+| `testTimeoutMinutes` || The maximum amount of time, in minutes, a single test should take. If this time is exceeded, the associated test will fail with a timeout error. If not given, the default value is `5`. |
+| `verbose` || Boolean value indicating whether to output more verbose logging information to standard output during a test run. If not given, the default is `false`. |
+
+## Test secrets
+
+The tests also expect to find several _secret_ parameters. While these can technically be added to `context.json`, it is recommended that you create environment variables and make them available to the test framework in a way that avoids committing them to your shell's command history or saving them in clear text on your filesystem. When set as environment variables, all secret parameters must be prefixed with `E2E_`. Parameter names are case-insensitive; they're only shown in uppercase here so they follow the common convention for environment variables, and to stand out as secrets.
+
+| Name | Required | Description |
+|------|----------|-------------|
+| `[E2E_]IOT_HUB_CONNECTION_STRING` | ✔ | Hub-scoped IoT Hub connection string that can be used to get/add/remove devices, deploy edge configurations, and get/update module twins. |
+| `[E2E_]EVENT_HUB_ENDPOINT` | ✔ | Connection string used to connect to the Event Hub-compatible endpoint of your IoT Hub, to listen for D2C events sent by devices or modules. |
+| `[E2E_]REGISTRIES__{n}__PASSWORD` || Password associated with a container registry entry in the `registries` array of `context.json`. `{n}` is the number corresponding to the (zero-based) array entry. For example, if you specified a single container registry in the `registries` array, the corresponding parameter would be `[E2E_]REGISTRIES__0__PASSWORD`. |
+| `[E2E_]ROOT_CA_PASSWORD` || The password associated with the root certificate specified in `rootCaCertificatePath`. |
+
+_Note: the definitive source for information about test parameters is `test/Microsoft.Azure.Devices.Edge.Test/helpers/Context.cs`._
+
+## Running the tests.
+
+With the test parameters and secrets in place, you can run all the end-to-end tests from the command line:
+
+```
+cd {repo root}
+dotnet test test/Microsoft.Azure.Devices.Edge.Test
```

Comment:
is it possible to pass context.json from dotnet test?  Also, when using this new test framework in VSTS pipeline for e2e test, we need to download the repo first, compile, update context json/environment varialbe, then run this command.  is it right? is it possible to run with compiled dll instead?

Reply:
> is it possible to pass context.json from dotnet test?

I think there's a way to do that; I investigated command line parameters a long time ago. Regardless, that's not how it's set up right now. The file needs to be in a specific location (same folder as the binary).

> we need to download the repo first, compile, update context json/environment varialbe, then run this command

Correct.

> is it possible to run with compiled dll instead?

Yes, you can also use `dotnet vstest` to run the tests.